sdcm/utils/loader_utils.py

```diff
@@ -117,6 +117,55 @@ def _run_all_stress_cmds(self, stress_queue, params):
 
         return stress_queue
 
+    @staticmethod
+    def parse_cs_user_profiles_param(cs_user_profiles):
+        """
+        The function parses user_profile parameter that defined in test yaml as:
+        ```
+            cs_user_profiles:
+                - scylla-qa-internal/custom_d1/rolling_upgrade_dataset.yaml,45m
+                - scylla-qa-internal/custom_d1/rolling_upgrade_dataset2.yaml,45m
+        ```
+        where first is user profile and second - how long the profile will be run (optional)
+        The goal is to create two lists:
+        - list of cs user profiles
+        - list of duration for every cs user profile (if defined)
+        The expected output of the function:
+            user_profiles = ['scylla-qa-internal/custom_d1/rolling_upgrade_dataset.yaml',
+                             'scylla-qa-internal/custom_d1/rolling_upgrade_dataset2.yaml']
+            duration_per_cs_profile = ['45m', '45m']
+        """
+        user_profiles_def = list(zip(*[profile.split(",") for profile in cs_user_profiles]))
+        user_profiles = list(user_profiles_def[0])
+        duration_per_cs_profile = None if len(user_profiles_def) == 1 else list(user_profiles_def[1])
+        return user_profiles, duration_per_cs_profile
+
+    def run_cs_user_profiles(self, cs_profiles: str | int | list, duration_per_cs_profile: str | list = None, stress_queue: list = None):  # pylint: disable=too-many-locals
+        stress_queue = stress_queue or []
```

Comment:
this is wrong in our context - if empty list is provided new list is used, while it is expected that original list was appended (like we do in `self._run_all_stress_cmds`)
Later in code where this method is used, return value is not used

Reply:
Not sure I understand.
`stress_queue` is mandatory in `_run_all_stress_cmds`. 
In this function it is optional. Do you suggest do it as mandatory argument?

Reply:
in case of `stress_queue` is provided with empty list, this line will anyway create a new list.
Our tests base on side-effect of changing provided list - which in that case won't happen.
To fix it do:
```
stress_queue = stress_queue if stress_queue is not None else []
```

Reply:
fixed